vpr/src/pack/greedy_clusterer.h

```diff
@@ -10,19 +10,41 @@
 
 #include <map>
 #include <unordered_set>
+#include <vector>
+#include "cluster_legalizer.h"
 #include "physical_types.h"
+#include "vtr_vector.h"
 
 // Forward declarations
 class AtomNetId;
 class AtomNetlist;
 class AttractionInfo;
-class ClusterLegalizer;
+class DeviceContext;
 class Prepacker;
+class SetupTimingInfo;
+class t_pack_high_fanout_thresholds;
+class t_pack_molecule;
 struct t_analysis_opts;
 struct t_clustering_data;
-struct t_pack_high_fanout_thresholds;
 struct t_packer_opts;
 
+/**
+ * @brief Struct to hold statistics on the progress of clustering.
+ *
+ * FIXME: These numbers only ever go up! This is a problem since some clusters
+ *        may be reclustered, leading to double counting. This is only a logging
+ *        bug, but should be thought about.
+ */
```

Comment:
We should fix this ... making the progress bar count finalized molecules makes more sense and will make this count accurate.